lib/experimental/fonts/font-library/class-wp-font-collection.php

```diff
@@ -21,41 +21,125 @@
 class WP_Font_Collection {
 
 	/**
-	 * Font collection configuration.
+	 * The unique slug for the font collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var string
+	 */
+	private $slug;
+
+	/**
+	 * The name of the font collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var string
+	 */
+	private $name;
+
+	/**
+	 * Description of the font collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var string
+	 */
+	private $description;
+
+	/**
+	 * Source of the font collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var string
+	 */
+	private $src;
+
+	/**
+	 * Array of font families in the collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var array
+	 */
+	private $font_families;
+
+	/**
+	 * Categories associated with the font collection.
 	 *
 	 * @since 6.5.0
 	 *
 	 * @var array
 	 */
-	private $config;
+	private $categories;
+
 
 	/**
 	 * WP_Font_Collection constructor.
 	 *
 	 * @since 6.5.0
 	 *
 	 * @param array $config Font collection config options.
-	 *  See {@see wp_register_font_collection()} for the supported fields.
-	 * @throws Exception If the required parameters are missing.
+	 *  {
+	 *      @type string $slug        The font collection's unique slug.
+	 *      @type string $name        The font collection's name.
+	 *      @type string $description The font collection's description.
+	 *      @type string $src         The font collection's source.
+	 *      @type array  $font_families An array of font families in the font collection.
+	 *      @type array  $categories The font collection's categories.
+	 *  }
 	 */
 	public function __construct( $config ) {
-		if ( empty( $config ) || ! is_array( $config ) ) {
-			throw new Exception( 'Font Collection config options is required as a non-empty array.' );
-		}
+		$this->is_config_valid( $config );
+
+		$this->slug          = $config['slug'] ?? '';
```

Comment:
Apparently using `??` is not agreed upon in Core: https://github.com/WordPress/gutenberg/pull/57688#discussion_r1456297013

Reply:
updated.